gcc/rust/hir/rust-ast-lower-block.h

```diff
@@ -120,6 +120,39 @@ class ASTLoweringIfBlock : public ASTLoweringBase
   bool terminated;
 };
 
+class ASTLoweringIfLetBlock : public ASTLoweringBase
```

Comment:
Any reason why you chose to add a new visitor class instead of simply implementing the function in `ASTLoweringIfBlock`? It seems you're not using new class members or anything, so it might be simpler to reuse this existing class.

Reply:
The only reason is that `translated` field in the `ASTLoweringIfBlock` is of type `HIR::IfExpr` and can't accomodate the `HIR::IfLetExpr`.

Reply:
Then I think it would make more sense to turn that into an `HIR::ExprWithBlock *` but @philberty might think your approach is better. I'm honestly fine with both, I just think it's easier for you to not have to reimplement an entire class/visitor :)

Reply:
Refactoring to use ExprWithBlock might be a really good idea to cleanup the code a bit here but lets do that as a separate piece since it will affect other code we can raise this as a good-first-pr refactor.